Brokerages/GDAX/GDAXBrokerage.Utility.cs

```diff
@@ -0,0 +1,140 @@
+
+            return token;
+        }
+
+        /// <summary>
+        /// Creates an auth token to sign a request
+        /// </summary>
+        /// <param name="body">the request body as json</param>
+        /// <param name="method">the http method</param>
+        /// <param name="url">the request url</param>
+        /// <returns></returns>
+        public AuthenticationToken GetAuthenticationToken(string body, string method, string url)
+        {
+            var token = new AuthenticationToken
+            {
+                Key = ApiKey,
+                Passphrase = _passPhrase,
+                Timestamp = Time.DateTimeToUnixTimeStamp(DateTime.UtcNow).ToString(System.Globalization.CultureInfo.InvariantCulture)
+            };
+
+            byte[] data = Convert.FromBase64String(ApiSecret);
+            var prehash = token.Timestamp + method + url + body;
+
+            byte[] bytes = Encoding.UTF8.GetBytes(prehash);
+            using (var hmac = new HMACSHA256(data))
+            {
+                byte[] hash = hmac.ComputeHash(bytes);
+                token.Signature = Convert.ToBase64String(hash);
+            }
+
+            return token;
+        }
+
+        private static string ConvertOrderType(Orders.OrderType orderType)
+        {
+            if (orderType == Orders.OrderType.Limit || orderType == Orders.OrderType.Market)
+            {
+                return orderType.ToString().ToLower();
+            }
+            else if (orderType == Orders.OrderType.StopMarket)
+            {
+                return "stop";
+            }
+
+            throw new Exception("Unsupported order type:" + orderType.ToString());
+        }
+
+        /// <summary>
+        /// Converts a product id to a symbol
+        /// </summary>
+        /// <param name="productId">gdax format product id</param>
+        /// <returns>Symbol</returns>
+        public static Symbol ConvertProductId(string productId)
+        {
+            return Symbol.Create(productId.Replace("-", ""), SecurityType.Forex, Market.GDAX);
+        }
+
+        /// <summary>
+        /// Converts a symbol to a product id
+        /// </summary>
+        /// <param name="symbol">Th symbol</param>
+        /// <returns>gdax product id</returns>
+        protected static string ConvertSymbol(Symbol symbol)
+        {
+            return symbol.Value.Substring(0, 3).ToLower() + "-" + symbol.Value.Substring(3, 3).ToLower();
+        }
+
+        private static Orders.OrderStatus ConvertOrderStatus(Messages.Order order)
+        {
+            if (order.FilledSize != 0 && order.FilledSize != order.Size)
+            {
+                return Orders.OrderStatus.PartiallyFilled;
+            }       
+            else if (order.Status == "open" || order.Status == "pending" || order.Status == "active")
+            {
+                return Orders.OrderStatus.Submitted;
+            }
+            else if (order.Status == "done" || order.Status == "settled")
```

Comment:
using string for the order status seems kind of error prone

Reply:
*undefined string

Reply:
Well, we could null check these strings, but what should we do if we encounter a null? I would say, given the context, we should throw an exception anyway.

I'm not aware that the project adheres strictly to the principle that these kind of strings should be constants. It might help to have constants here.